app/controllers/tasks_controller.rb

```diff
@@ -1,80 +1,133 @@
 class TasksController < ApplicationController
 
   def index
-    @all_tasks = Task.all   # NOTE TO SELF: "task" is actually a Task::ActiveRecord_Relation object, but responds to Array methods
-    completed_instance_var
+    @all_tasks = Task.all
+      # NOTE TO SELF: "@all_tasks" is actually a Task::ActiveRecord_Relation object, but responds to array methods
 
-    render :index   # for clarity, you don't actually need this in this case
+    render :index
+      # NOTE TO SELF: this is for clarity, you don't actually need this in this case
+  end
+
+  def completed_status
+    @task = Task.find(params[:id])
+    if @task[:completed_at].nil?
+      Task.completed(params[:id])
+    else
+      Task.not_completed(params[:id])
+    end
+
```

Comment:
I like the way that you've decided to encapsulate the logic here. The only issue with this is that you're doing the same lookup multiple times. You're going to the database to find the task on line 12, and you're doing the same task lookup within the model in the `completed` and `not_completed` methods. Database lookups can be the cause of inefficiencies in your code, so it's good to be sure that you only have to look up exactly as much information as you need.